test/unittests/utils/test_filemanip.py

```diff
@@ -179,6 +181,47 @@ def test_zip_unzip_nii(tmp_path):
     assert line == "Test"
 
 
+@pytest.fixture
+def test_image(case):
+    shapes = {
+        "3d": (5, 6, 7),
+        "4d_dummy": (5, 6, 7, 1),
+        "4d": (5, 6, 7, 8),
+        "5d": (5, 6, 7, 8, 9),
+    }
+    data = np.zeros(shapes[case])
+    data[2:4, 1:5, 3:6] = 1
+    affine = np.diag((4, 3, 2, 1))
+    return nib.Nifti1Image(data, affine=affine)
+
+
+@pytest.mark.parametrize("case", ["3d", "4d_dummy", "4d", "5d"])
+def test_load_img_3d(tmp_path, case, test_image):
+    from clinica.utils.filemanip import load_img_3d
+
+    with pytest.raises(
+        FileNotFoundError,
+        match="No such file or no access: 'foo'",
+    ):
+        load_img_3d("foo")
+    (tmp_path / "sub-01").mkdir()
+    (tmp_path / "sub-01" / "ses-M000").mkdir()
```

Comment:
```suggestion
    (tmp_path / "sub-01" / "ses-M000").mkdir(parents=True)
```

Reply:
I wonder whether we really need a BIDS-like layout to test this function, instead of just creating a nifti file with some random name under `tmp_path`?

Reply:
I initially thought the same, but you do need a BIDS-like layout because of the call to `get_subject_id` (`image_id = get_subject_id(str(image_path))`), which parses the `image_path` with the underlying assumption that it is a BIDS compliant name.

Reply:
Then I believe this piece of logic should go elsewhere:

```
       image_id = get_subject_id(str(image_path))
        error_msg = (
            f"Clinica does not handle {dim}D volumes for {image_id.replace('_', ' | ')}"
        )
```

which would make `load_img_3d` more generic (why not `load_volume` actually?).

Imo, there is nothing particulary BIDS specific to this loader apart from the error handling logic, which could be deported upstream (where the BIDS context exists) to reduce coupling.

Also `NotImplementedError` is probably not the exception type we want to raise. It's mostly used for ABCs to signal uncomplete class derivation. Perhaps we should use a good old `ValueError` instead.

Reply:
I agree with your points. I proposed an implementation in 19a794e
Let me know if this is more or less what you had in mind.

Reply:
Perfecto 👌